src/main/java/org/jabref/model/texparser/CrossingKeysResult.java

```diff
@@ -0,0 +1,157 @@
+    private final BibDatabase masterDatabase;
+    private final List<String> unresolvedKeys;
+    private final BibDatabase newDatabase;
+    private int crossRefsCount;
+
+    public CrossingKeysResult(TexParserResult texParserResult, BibDatabase masterDatabase) {
+        this.texParserResult = texParserResult;
+        this.masterDatabase = masterDatabase;
+        this.unresolvedKeys = new ArrayList<>();
+        this.newDatabase = new BibDatabase();
+        this.crossRefsCount = 0;
+    }
+
+    public TexParserResult getTexParserResult() {
+        return texParserResult;
+    }
+
+    public BibDatabase getMasterDatabase() {
+        return masterDatabase;
+    }
+
+    public List<String> getUnresolvedKeys() {
+        return unresolvedKeys;
+    }
+
+    public BibDatabase getNewDatabase() {
+        return newDatabase;
+    }
+
+    public int getCrossRefsCount() {
+        return crossRefsCount;
+    }
+
+    /**
+     * Return the citations map from the TexParserResult object.
+     */
+    public Map<String, Set<Citation>> getCitations() {
+        return texParserResult.getCitations();
+    }
+
+    /**
+     * Return a set of strings with the keys of the citations map from the TexParserResult object.
+     */
+    public Set<String> getCitationsKeySet() {
+        return texParserResult.getCitations().keySet();
+    }
+
+    /**
+     * Return the master database in a set, for comparing two objects.
+     */
+    public Set<BibEntry> getMasterDatabaseSet() {
+        return new HashSet<>(masterDatabase.getEntries());
+    }
+
+    /**
+     * Get if an entry with the given key is present in the master database.
+     */
+    public Optional<BibEntry> getEntryMasterDatabase(String key) {
+        return masterDatabase.getEntryByKey(key);
+    }
+
+    /**
+     * Add an unresolved key to the list.
+     */
+    public void addUnresolvedKey(String key) {
+        unresolvedKeys.add(key);
+    }
+
+    /**
+     * Return the new database in a set, for comparing two objects.
+     */
+    public Set<BibEntry> getNewDatabaseSet() {
+        return new HashSet<>(newDatabase.getEntries());
+    }
+
+    /**
+     * Check if an entry with the given key is present in the new database.
+     */
+    public boolean checkEntryNewDatabase(String key) {
+        return newDatabase.getEntryByKey(key).isPresent();
+    }
+
+    /**
+     * Add 1 to cross references counter.
+     */
+    public void increaseCrossRefsCount() {
+        crossRefsCount++;
+    }
+
+    /**
+     * Insert into the database a clone of an entry with the given key. The cloned entry has a new unique ID.
+     */
+    public void insertEntry(String key) {
+        insertEntry(masterDatabase.getEntryByKey(key).get());
+    }
+
+    /**
+     * Insert into the database a clone of the given entry. The cloned entry has a new unique ID.
+     */
+    public void insertEntry(BibEntry entry) {
+        BibEntry clonedEntry = (BibEntry) entry.clone();
```

Comment:
Do we really need to clone the entry? That is usually not necessary and might lead to confusions (e.g. when you modify the cloned entry without realizing that it is a clone and thus does not affect the original entry).